### PR TITLE
Fix gRPC Python issue since switch to 0.13.

### PR DESCRIPTION
**Pull request by michael-berlin**
@enisoc 

See the commit messages for the details.

I haven't tested it on Travis yet. I'll post here once that's done.

[Review on Reviewable](https://reviewable.io/reviews/youtube/vitess/1541)

### CODE REVIEW

**Comment by enisoc** [comment]
LGTM. Have you tried `make docker_bootstrap` as well?

**Comment by enisoc** [comment]
Some of your travis shards failed like this:

```
TypeError: __init__() got an unexpected keyword argument 'syntax'
```

This is the error I saw before, when the older version of protobuf was being used. The `syntax=` keyword arg for the google.protobuf.Descriptor constructor is new. My guess for why only some shards failed was that they somehow got delayed and ended up using the cache populated by one of the other shards.

**Comment by michael-berlin** [comment]
That analysis makes sense. But it's still weird that the non-cached test run fails, but the one using the cache succeeds?!

I can add a Python script tomorrow to further debug it.


```make docker_bootstrap``` is also failing :( It looks like Debian doesn't play so nice and upgrades the "six" module automatically:

```
+ pip install .
[...]
    Installed /tmp/pip-j5H0Na-build/.eggs/enum34-1.1.2-py2.7.egg
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-j5H0Na-build/setup.py", line 242, in <module>
        setuptools.setup(**setup_arguments)
      File "/usr/lib/python2.7/distutils/core.py", line 111, in setup
        _setup_distribution = dist = klass(attrs)
      File "/usr/local/lib/python2.7/dist-packages/setuptools/dist.py", line 269, in __init__
        self.fetch_build_eggs(attrs['setup_requires'])
      File "/usr/local/lib/python2.7/dist-packages/setuptools/dist.py", line 313, in fetch_build_eggs
        replace_conflicting=True,
      File "/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 827, in resolve
        dist = best[req.key] = env.best_match(req, ws, installer)
      File "/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 1065, in best_match
        dist = working_set.find(req)
      File "/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 702, in find
        raise VersionConflict(dist, req)
    pkg_resources.VersionConflict: (six 1.8.0 (/usr/lib/python2.7/dist-packages), Requirement.parse('six>=1.10'))
    
    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-j5H0Na-build
The command '/bin/sh -c cd /vt/dist &&     INSTALL_GRPC_PHP=`php-config --extension-dir` $VTTOP/travis/install_grpc.sh &&     echo 'extension=grpc.so' > /etc/php5/cli/conf.d/20-grpc.ini &&     rm -rf /vt/dist/grpc' returned a non-zero code: 1
make: *** [docker_bootstrap] Error 1
```

A similar issue is also reported on the grpc project: https://github.com/grpc/grpc/issues/5518

Proposed solution is to force upgrade the ```six``` package:
```sh
pip install --ignore-installed six
```

**Comment by enisoc** [comment]
I was saying that the cached one fails. In my case it was because the right version of protobuf was being installed outside dist/grpc/usr/local. So that version disappears after going through a save->restore cycle.

**Comment by michael-berlin** [comment]
@enisoc PTAL

```make docker_bootstrap``` passes now and Travis as well.

File: docker/bootstrap/Dockerfile.common [hunk 1]
```diff
@@ -1,5 +1,8 @@
 FROM golang:1.5
 
+# Use all cores for parallel make (load bound to number of cores as well).
+ENV MAKEFLAGS -j$(nproc) -l$(nproc)
```
Comment (enisoc) [inline docker/bootstrap/Dockerfile.common]: I think this hard-codes the $(nproc) value at the time of building the image, and that value will take effect for everything you do with that image on any machine later. I'd prefer that this env var be set only for the install_grpc.sh script - either within the script, or at the start of the RUN directive.

**Comment by enisoc** [comment]
LGTM with one comment.

File: docker/bootstrap/Dockerfile.common (hunk 1 above)
Comment (michael-berlin) [inline docker/bootstrap/Dockerfile.common]: DONE

**Comment by enisoc** [comment]
LGTM

**Pull request merged**